src/bun.js/test/pretty_format.zig

```diff
@@ -1986,4 +1951,141 @@ pub const JestPrettyFormat = struct {
             };
         }
     };
+
+    fn printAsymmetricMatcherPromisePrefix(flags: expect.Expect.Flags, matcher: anytype, writer: anytype) void {
+        if (flags.promise != .none) {
+            switch (flags.promise) {
+                .resolves => {
+                    matcher.addForNewLine("promise resolved to ".len);
+                    writer.writeAll("promise resolved to ");
+                },
+                .rejects => {
+                    matcher.addForNewLine("promise rejected to ".len);
+                    writer.writeAll("promise rejected to ");
+                },
+                else => {},
+            }
+        }
+    }
+
+    pub fn printAsymmetricMatcher(
+        // the Formatter instance
+        this: anytype,
+        comptime Format: anytype,
+        /// The WrappedWriter
+        writer: anytype,
+        /// The raw writer
+        writer_: anytype,
+        /// Buf used to print strings
+        name_buf: [512]u8,
+        value: JSValue,
+        comptime enable_ansi_colors: bool,
+    ) bool {
+        _ = Format;
+
+        if (value.as(expect.ExpectAnything)) |matcher| {
+            if (matcher.flags.not) {
+                this.addForNewLine("NotAnything".len);
+                writer.writeAll("NotAnything");
+            } else {
+                this.addForNewLine("Anything".len);
+                writer.writeAll("Anything");
+            }
+        } else if (value.as(expect.ExpectAny)) |matcher| {
+            const constructor_value = expect.ExpectAny.constructorValueGetCached(value) orelse return true;
+
+            printAsymmetricMatcherPromisePrefix(matcher.flags, this, writer);
+            if (matcher.flags.not) {
+                this.addForNewLine("Any<".len);
```

Comment:
`"NotAny<"` and `"Any<"` need to swap places

Reply:
good catch!